### ISSUE TITLE
[FEATURE_REQUEST] Explicit check for `securityContext.allowPrivilegeEscalation=false` on containers

### ISSUE THREAD

**Issue by AlanMasciangelo** [stackrox/kube-linter#668]
**Description of the problem/feature request**
Add a check or option to check if containers explicitly set allowPrivilegeEscalation to false. Explicitly setting this to false is generally regarded as good security practice.

**Description of the existing behavior vs. expected behavior**

The existing `privilege-escalation-container` doesn't trigger on this condition, I imagine intentionally since allowPrivilegeEscalation is determined by a variety of factors and can be restricted in other ways.

**Additional context**

**Comment by janisz** [comment]
@AlanMasciangelo - CEL expression solution using the new template:

- https://github.com/stackrox/kube-linter/pull/1012

```yaml
customChecks:
  - name: explicit-privilege-escalation-false
    template: cel-expression
    params:
      check: |
        has(object.spec.containers) && 
        object.spec.containers.exists(c, 
          !has(c.securityContext.allowPrivilegeEscalation) || 
          c.securityContext.allowPrivilegeEscalation != false) ?
        "Container should explicitly set allowPrivilegeEscalation: false" : ""
    scope:
      objectKinds: [DeploymentLike]
```

This checks that all containers explicitly set `allowPrivilegeEscalation` to false.
